lib/Adapter/TolerantParser/Fixer/MemberEmptyLineFixer.php

```diff
@@ -0,0 +1,187 @@
+<?php
+
+namespace Phpactor\CodeBuilder\Adapter\TolerantParser\Fixer;
+
+use Microsoft\PhpParser\Node;
+use Microsoft\PhpParser\Node\ClassConstDeclaration;
+use Microsoft\PhpParser\Node\MethodDeclaration;
+use Microsoft\PhpParser\Node\PropertyDeclaration;
+use Microsoft\PhpParser\Node\TraitUseClause;
+use Microsoft\PhpParser\Parser;
+use Phpactor\CodeBuilder\Adapter\TolerantParser\Util\NodeHelper;
+use Phpactor\CodeBuilder\Domain\StyleProposer;
+use Phpactor\CodeBuilder\Domain\TextEdit;
+use Phpactor\CodeBuilder\Domain\TextEdits;
+use Phpactor\CodeBuilder\Util\TextFormat;
+use Phpactor\CodeBuilder\Util\TextUtil;
+
+class MemberEmptyLineFixer implements StyleProposer
+{
+    private const META_SUCCESSOR = 'successor';
+    private const META_NODE_CLASS = 'class';
+    private const META_FIRST = 'first';
+    private const META_PRECEDING_BLANK_LINES = 'preceding_blank_lines';
+    private const META_PRECEDING_BLANK_START = 'blank_start';
+    private const META_PRECEDING_BLANK_LENGTH = 'blank_length';
+    private const META_INDENTATION = 'indentation';
+    private const META_IS_METHOD = 'is_method';
+    private const META_HAS_DOCBLOCK = 'has_docblock';
+
+    /**
+     * @var Parser
+     */
+    private $parser;
+
+    /**
+     * @var TextFormat
+     */
+    private $textFormat;
+
+    public function __construct(?Parser $parser = null, ?TextFormat $textFormat = null)
+    {
+        $this->parser = $parser ?: new Parser();
+        $this->textFormat = $textFormat ?: new TextFormat();
+    }
+
+    public function propose(string $text): TextEdits
+    {
+        $node = $this->parser->parseSourceFile($text);
+
+        $membersMeta = $this->gatherMetadata($node, [
+            TraitUseClause::class,
+            ClassConstDeclaration::class,
+            PropertyDeclaration::class,
+            MethodDeclaration::class,
+        ]);
+
+        return TextEdits::fromTextEdits($this->textEdits($membersMeta));
+    }
+
+    private function gatherMetadata(Node $node, array $nodeTypes): array
+    {
+        $nodesMeta = [];
+        $nodes = [];
+
+        $previousNodeClass = null;
+        foreach (NodeHelper::nodesOfTypes($nodeTypes, $node) as $node) {
+            assert($node instanceof Node);
+            $meta = [
+                self::META_SUCCESSOR => false,
+                self::META_NODE_CLASS => get_class($node),
+                self::META_FIRST => false,
+                self::META_PRECEDING_BLANK_LINES => $this->countBlankLines($node->getLeadingCommentAndWhitespaceText()),
+                self::META_PRECEDING_BLANK_START => $node->getFullStart(),
+                self::META_PRECEDING_BLANK_LENGTH => $this->blankLength($node),
+                self::META_INDENTATION => $this->indentation($node),
+                self::META_IS_METHOD => $node instanceof MethodDeclaration,
+                self::META_HAS_DOCBLOCK => TextUtil::hasDocblock($node->getLeadingCommentAndWhitespaceText()),
```

Comment:
You can retrieve the docblock directly from the `$node` and just check if it's null or not.
Might avoid to add another method in the `TextUtil` class

Reply:
Done -- didn't realise that method existed. It's _potentially_ more expensive then the TextUtil method however as it parses the doc comment (but only as far as saying it is a docblock token, it doesn't parse the tags).